### PR TITLE
Implemented the wait flag.

### PR DESCRIPTION
**Pull request by mikeecb**
Implemented the wait flag. We now send packets and listen for incoming packets in separate threads.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -26,38 +28,80 @@ fn main() {
     let host = args[args.len() - 1].clone();
     let ip_address = IpAddr::V4(Ipv4Addr::from_str(&host).unwrap());
 
+    let mut opts = Options::new();
+    opts.optopt("i", "", "Wait wait seconds between sending each packet. The default is to wait for one second between each packet. The wait time may be fractional, but only the super-user may specify values less than 0.1 second.  This option is incompatible with the -f option.", "wait");
```
Comment (SlavGradev) [inline src/main.rs]: Wait wait seconds?

File: src/main.rs [hunk 2]
```diff
@@ -26,38 +28,80 @@ fn main() {
...
+                // TODO: Maybe we should also display the usage
+                Err(_) => default_time_between_packets 
+            }
+        },
+        None => default_time_between_packets
+    };
+    let time_between_packets = Duration::from_secs(time_between_packets_secs);
+
+    // Incoming packet iterator
+    let mut iter = ipv4_packet_iter(&mut t_receiver);
```
Comment (SlavGradev) [inline src/main.rs]: Can you move this iterator just before the loop for incoming packets

File: src/main.rs (hunk 1 above)
Comment (mikeecb) [inline src/main.rs]: The second "wait" should be "_wait_". I copied and pasted this from `man ping`.

File: src/main.rs (hunk 2 above)
Comment (mikeecb) [inline src/main.rs]: Sure

**Pull request merged**